src/main/java/org/openlmis/buq/service/buq/BottomUpQuantificationService.java

```diff
@@ -183,13 +182,20 @@ private BottomUpQuantification prepareBottomUpQuantification(FacilityDto facilit
 
   private void prepareLineItems(
       BottomUpQuantification bottomUpQuantification,
-      List<ApprovedProductDto> approvedProducts) {
+      List<ApprovedProductDto> approvedProducts,
+      ProcessingPeriodDto processingPeriod) {
     List<BottomUpQuantificationLineItem> bottomUpQuantificationLineItems = new ArrayList<>();
     for (ApprovedProductDto product : approvedProducts) {
       BottomUpQuantificationLineItem lineItem = new BottomUpQuantificationLineItem();
       lineItem.setBottomUpQuantification(bottomUpQuantification);
       lineItem.setOrderableId(product.getOrderable().getId());
 
+      Integer annualAdjustedConsumption = AnnualAdjustedConsumptionCalculator.calculate(
```

Comment:
Something confusing here.
The `bottomUpQuantification.getProcessingPeriodId()` and `processingPeriod` refer to the the same period.
The `bottomUpQuantificationRepository.getRequisitionLineItemsData()` returns line items already in the period.
The `AnnualAdjustedConsumptionCalculator.calculate()` performs the filtering by the same period again.

Was is intended? 

On the other hand, this sum inside `AnnualAdjustedConsumptionCalculator#calculate` looks like could be done in DB and have something like: `bottomUpQuantificationRepository.getSumOfRequisitionLineItemsAdjustedConsumptions(orderableId, facilityId, processingPeriodId)`

Reply:
Changed to use an aggregate function.